algorithm/MovieLens.R

```diff
@@ -10,48 +11,49 @@ library(sampling)
 #
 # Datasets are acquired from website https://grouplens.org/datasets/movielens
 setwd("Documents/R/datasets/ml-100k")
-data <- read.table("u.data")[ , -4]
-names(data) <- c("UserID", "MovieID", "Rating")
+rating <- read.table("u.data")[ , -4]
+names(rating) <- c("UserID", "MovieID", "Rating")
 
 # Compile the function "MovieLensKNN"
-MovieLensKNN <- function(userID, movieID, n, K) {
-  targetUsers <- which(data$UserID == userID)
-  relativeMovies <- sample(targetUsers, min(length(targetUsers), n))
+MovieLensKNN <- function(userID, movieID, n, k) {
+  targetUserIndexes <- which(rating$UserID == userID)
+  sampleIndexes <- sample(targetUserIndexes, min(length(targetUserIndexes), n))
 
-  relativeMoviesID <- data$MovieID[relativeMovies]
-  targetMoviesID <- movieID
+  sampleMovieIDs <- rating$MovieID[sampleIndexes]
+  targetMovieID <- movieID
 
-  trainingMovies <- which(data$MovieID == targetMoviesID)
-  trainingUsers <- data$UserID[trainingMovies[-1]]
+  targetMovieIndexes <- which(rating$MovieID == targetMovieID)
+  targetUser <- which(rating$UserID[targetMovieIndexes] == userID)
+  trainingUsers <- rating$UserID[targetMovieIndexes][-targetUser]
```

Comment:
It would be clearer if you could add the following comment before the code block:

    # Prepare training set (target user excluded)